In [1]:
import numpy as np
import pandas as pd
pd.set_option('display.max_rows', 200)
import matplotlib.pyplot as plt
#import seaborn as sb
%matplotlib inline

In [2]:
data = pd.read_excel('../Ann_med/med1.xls')

In [3]:
data.head(5)

Международное непатентованное наименование  Код АТХ                Группа  \
0                                        NaN      NaN                   NaN   
1                                        NaN      NaN                   NaN   
2                                  Десфлуран  N01AB07  Средства для наркоза   
3                          Динитрогена оксид  N01AX13  Средства для наркоза   
4                                  Изофлуран  N01AB06  Средства для наркоза   

                       Латинское наименование Код АТХ.1  \
0                               I. Анестетики       NaN   
1                        Средства для наркоза       NaN   
2  Suprane 240 ml N1 жидкость д/ингал.наркоза   N01AB07   
3                                               N01AX13   
4  Aerrane 100 ml N6 жидкость д/ингал.наркоза   N01AB06   

  Описание назначения ЛС                          Лекарственная форма, доза  \
0                    NaN                                                NaN   
1                    NaN                                                NaN   
2   Средства для наркоза                      Жидкость для ингаляций, 240мл   
3   Средства для наркоза                                Газ сжатый /баллоны   
4   Средства для наркоза  Жидкость для ингаляционной анестезии, флаконы ...   

   DDD Категория Категория VEN  
0  NaN       NaN           NaN  
1  NaN       NaN           NaN  
2  нет       ПКУ             V  
3  нет     ЖНВЛП             V  
4  нет       NaN             E

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1102 entries, 0 to 1101
Data columns (total 10 columns):
Международное непатентованное наименование    1080 non-null object
Код АТХ                                       1100 non-null object
Группа                                        979 non-null object
Латинское наименование                        1102 non-null object
Код АТХ.1                                     952 non-null object
Описание назначения ЛС                        979 non-null object
Лекарственная форма, доза                     953 non-null object
DDD                                           904 non-null object
Категория                                     647 non-null object
Категория VEN                                 946 non-null object
dtypes: object(10)
memory usage: 86.2+ KB


В таблице 1102 записи, при этом если убрать пропуски остается только 608 записей

In [5]:
data.dropna(inplace=True)

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 608 entries, 2 to 1100
Data columns (total 10 columns):
Международное непатентованное наименование    608 non-null object
Код АТХ                                       608 non-null object
Группа                                        608 non-null object
Латинское наименование                        608 non-null object
Код АТХ.1                                     608 non-null object
Описание назначения ЛС                        608 non-null object
Лекарственная форма, доза                     608 non-null object
DDD                                           608 non-null object
Категория                                     608 non-null object
Категория VEN                                 608 non-null object
dtypes: object(10)
memory usage: 52.2+ KB


Вот уникальные записи.. запарсить это очень сложно. Допустим таблетки я еще разных форм вытащу и посчитаю, это не проблема, но дальше идет мега список из всяких растворов и каждый раз это новое название : (  см ниже

In [7]:
data["Лекарственная форма, доза"].value_counts()

Таблетки 25мг                                                                                                      9
Таблетки 500мг                                                                                                     9
Таблетки 10мг                                                                                                      9
Таблетки 50мг                                                                                                      8
Таблетки 5мг                                                                                                       8
Таблетки 100мг                                                                                                     8
Таблетки 200мг                                                                                                     7
Таблетки 40мг                                                                                                      7
Порошок для приготовления раствора для инъекций фл. 1г          

вот полный список уникальных записей.. есть ли идеи, как это можно группировать?

In [8]:
data["Лекарственная форма, доза"].unique()

array(['Жидкость для ингаляций, 240мл', 'Газ сжатый /баллоны',
       'Раствор для инъекций амп. 5% 2,0мл',
       'Раствор для инъекций амп. 20% 5,0мл',
       'Эмульсия для внутривенного введения амп. 1% 20мл',
       'Эмульсия для внутривенного  введения амп. 1% 50мл',
       'Эмульсия для внутривенного  введения амп. 20мг\\мл-50 мл',
       'Жидкость для ингаляций фл.250мл',
       'Раствор для инъекций амп.0,5% 20,0мл',
       'Раствор для инъекций амп.0,5% 4,0мл',
       'Раствор для инъекций амп.0,5% 5,0мл',
       'Раствор для инъекций амп.2% 2,0мл',
       'Раствор для инъекций 0,25% 200,0',
       'Раствор для инъекций 0,5% 200,0',
       'Раствор для инъекций 0,25% 400,0',
       'Раствор для инъекций 0,5% 400,0',
       'Раствор для инъекций 2мг/мл 100мл',
       'Раствор для инъекций амп. 2мг/мл 20мл',
       'Раствор для инъекций амп. 7,5мг/мл 10,0мл',
       'Раствор для инъекций амп. 10мг/мл 10,0мл',
       'Раствор для инъекций амп. 5мг/мл 10,0мл',
       'Раствор для 

In [9]:
data[u'Лекарственная форма, доза'] = data[u'Лекарственная форма, доза'].str.split(" ")
data[u'Лекарственная форма, доза'] 

2                      [Жидкость, для, ингаляций,, 240мл]
3                                 [Газ, сжатый, /баллоны]
5               [Раствор, для, инъекций, амп., 5%, 2,0мл]
6              [Раствор, для, инъекций, амп., 20%, 5,0мл]
7       [Эмульсия, для, внутривенного, введения, амп.,...
8       [Эмульсия, для, внутривенного, , введения, амп...
9       [Эмульсия, для, внутривенного, , введения, амп...
10                   [Жидкость, для, ингаляций, фл.250мл]
12             [Раствор, для, инъекций, амп.0,5%, 20,0мл]
13              [Раствор, для, инъекций, амп.0,5%, 4,0мл]
19              [Раствор, для, инъекций, амп.0,5%, 5,0мл]
20                [Раствор, для, инъекций, амп.2%, 2,0мл]
21                 [Раствор, для, инъекций, 0,25%, 200,0]
22                  [Раствор, для, инъекций, 0,5%, 200,0]
23                 [Раствор, для, инъекций, 0,25%, 400,0]
24                  [Раствор, для, инъекций, 0,5%, 400,0]
25                [Раствор, для, инъекций, 2мг/мл, 100мл]
26           [